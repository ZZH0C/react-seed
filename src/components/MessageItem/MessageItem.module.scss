.media_message{
  position: relative;
  margin: 1rem;
  zoom: 1;
  overflow: hidden;
}

.delete_button{
  background-color: #f2f4f8;
border-radius: 0.5rem;
  color: #99a1b3;
  width: 11.5rem;
  height: 2rem;
padding: 0;
  margin: 0;
  border: none;
}

.item_name{
  margin: 0.3rem 0;
  display: flex;
  justify-content: space-between;
  padding-right: 0.2rem;
}

.delete_button:hover {
  box-shadow: 0 0.2rem 0.2rem 0 rgba(0, 0, 0, 0.5);
}

.delete_button:active {
  box-shadow: inset 0 0.2rem 0.2rem 0 rgba(0, 0, 0, 1);
}