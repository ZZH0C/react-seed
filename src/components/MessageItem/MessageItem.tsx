import React from 'react';
import { Link } from 'react-router-dom';

interface MessageItemProps extends React.HTMLAttributes<HTMLElement> {
  fromWho: string;
  messageTitle: string;
  messageSnippet: string;
  messageDate: string;
  messageId: string;
}

export const MessageItem: React.FC<MessageItemProps> = ({
  fromWho,
  messageTitle,
  messageSnippet,
  messageDate,
  messageId,
}) => {
  return (
    <section className="media">
      <div className="media-body">
        <div className="userInfo clearfix">
          <span>from: {fromWho}</span>
          <div className="commentsAndTime pull-right">
            <span>
              <i className="icon-clock" />
              <time className="timeago" dateTime={messageDate}>
                {messageDate}
              </time>
            </span>
          </div>
        </div>
        <div className="itemName">
          <Link
            to={{
              pathname: '/message',
              state: messageId,
            }}
          >
            <span className="media-heading" title="Message Title">
              {messageTitle}
            </span>
          </Link>
          <span className="status pull-right" />
        </div>
        <p>{messageSnippet} </p>
      </div>
    </section>
  );
};
