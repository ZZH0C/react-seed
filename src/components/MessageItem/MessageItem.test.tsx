import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { MessageItem } from './MessageItem';

describe('components/ButtonLink', () => {
  const mockProps = {
    from: 'SOME_USER',
    date: '11-03-1996',
    id: 'FAKE_ID',
    title: 'SOME_TITLE',
    snippet: 'SOME_TEXT',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(
        <MessageItem
          fromWho={mockProps.from}
          messageDate={mockProps.date}
          messageId={mockProps.id}
          messageSnippet={mockProps.snippet}
          messageTitle={mockProps.title}
        />,
      ),
    ).toMatchSnapshot();
  });
});
