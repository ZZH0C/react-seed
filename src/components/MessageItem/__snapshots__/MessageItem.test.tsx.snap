// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<section
  className="media"
>
  <div
    className="media-body"
  >
    <div
      className="userInfo clearfix"
    >
      <span>
        from: 
        SOME_USER
      </span>
      <div
        className="commentsAndTime pull-right"
      >
        <span>
          <i
            className="icon-clock"
          />
          <time
            className="timeago"
            dateTime="11-03-1996"
          >
            11-03-1996
          </time>
        </span>
      </div>
    </div>
    <div
      className="itemName"
    >
      <Link
        to={
          Object {
            "pathname": "/message",
            "state": Object {
              "id": "FAKE_ID",
              "token": "",
            },
          }
        }
      >
        <span
          className="media-heading"
          title="Message Title"
        >
          SOME_TITLE
        </span>
      </Link>
      <span
        className="status pull-right"
      />
    </div>
    <p>
      SOME_TEXT
       
    </p>
  </div>
</section>
`;
